src/Filecoin/Address.h

```diff
@@ -0,0 +1,104 @@
+// Copyright © 2017-2020 Trust.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#pragma once
+
+#include "../PublicKey.h"
+
+#include <array>
+#include <cstdint>
+#include <optional>
+#include <string>
+#include <vector>
+
+namespace TW::Filecoin {
+
+class Address {
+  public:
+    enum class Type : uint8_t {
+        ID = 0,
+        SECP256K1 = 1,
+        ACTOR = 2,
+        BLS = 3,
+    };
+
+    /// Address data with address type prefix.
+    Data bytes;
+
+    /// Determines whether a collection of bytes makes a valid address.
+    static bool isValid(const Data& data);
+
+    /// Determines whether a string makes a valid encoded address.
+    static bool isValid(const std::string& string);
+
+    /// Initializes an address with a string representation.
+    explicit Address(const std::string& string);
+
+    /// Initializes an address with a collection of bytes.
+    explicit Address(const Data& data);
+
+    /// Initializes an address with a secp256k1 public key.
+    explicit Address(const PublicKey& publicKey);
+
+    /// Returns a string representation of the address.
+    [[nodiscard]] std::string string() const;
+
+    /// Returns the type of an address.
+    Type type() const { return *getType(bytes[0]); }
```

Comment:
What if getType() returns an empty value?  Wouldn't it be safer/simpler if a Type::Invalid enum value is used, without any optional?